README.md

```diff
@@ -280,6 +280,53 @@ change to a direct dependency. This process applies to development dependencies
 If you wish to re-pin all the dependencies, run `make refresh_all_requirements`.  It is advisable to do a full
 test-deploy-test cycle after this (the test after the deploy is required to test the lambdas).
 
+#### Logging conventions
+
+1.  Always use a module-level logger, call it `logger` and initialize it as follows:
+
+    ```python
+    logger = logging.getLogger(__name__)
+    ```
+
+2.  Do not configure logging at module scope. It should be possible to import any module without side-effects on 
+    logging. The `dss.logging` module contains functions that configure logging for this application, its Lambda 
+    functions and unit tests.
+    
+3.  When logging a message, pass either
+ 
+    * an f-string as the first and only positional argument or
```

Comment:
I'm inclined to just ban f-strings because the modifying a formatting string to take more arguments will cause unnecessary churn or people might forget to switch to interpolation.

Reply:
I thought I understood this comment but now I'm not so sure.

f-strings *are* a kind of interpolation. Do you mean "switch to % interpolation"? There really is no reason to perform that switch.

Reply:
My understanding is that if you have f-strings with any string formatting happening, then the output could contain `%`, which would cause logging to freak out.  It seems like it's just risky.

In your example:

```
logger.info(f"Foo is {foo} and bar is {bar}")
```

if `foo` is `%s`, wouldn't logging complain that there aren't enough arguments to format the output?

Reply:
So I am suggesting that we just use % interpolation everywhere.

Reply:
> wouldn't logging complain

Nope: https://github.com/HumanCellAtlas/data-store/issues/829#issuecomment-358455378

> So I am suggesting that we just use % interpolation everywhere.

But why?

Reply:
Maybe I should add a section explaining why it is safe to use an f"" string as the first argument as long as you don't pass any additional positional arguments.

Reply:
didn't we have this discussion https://github.com/HumanCellAtlas/data-store/pull/859#discussion_r163368764  already in a previous comment string or is this something different?

Reply:
Same thing. My bad.